app/ts/background/background.ts

```diff
@@ -192,21 +192,36 @@ async function handleRPCRequest(
 ): Promise<RPCReply> {
 	const maybeParsedRequest = EthereumJsonRpcRequest.safeParse(request)
 	const forwardToSigner = !settings.simulationMode && !request.usingInterceptorWithoutSigner
+	const getForwardingMessage = (request: SendRawTransaction | SendTransactionParams | WalletAddEthereumChain | EthGetStorageAtParams) => {
+		if (!forwardToSigner) throw new Error('Should not forward to signer')
+		return { forward: true as const, ...request }
+	}
+
 	if (maybeParsedRequest.success === false) {
 		console.log(request)
 		console.warn(maybeParsedRequest.fullError)
+		const maybePartiallyParsedRequest = SupportedEthereumJsonRpcRequestMethods.safeParse(request)
+		if (maybePartiallyParsedRequest.success === false) {
+			// the method is some method that we are not supporting, forward it to the wallet if signer is available
+			if (forwardToSigner) return { forward: true as const, unknownMethod: true, ...request }
+			return {
+				method: request.method,
+				error: {
+					message: `Failed to parse RPC request: ${ JSON.stringify(request) }`,
+					data: maybeParsedRequest.fullError === undefined ? 'Failed to parse RPC request' : maybeParsedRequest.fullError.toString(),
+					code: METAMASK_ERROR_FAILED_TO_PARSE_REQUEST,
+				}
+			}
+		}
 		return {
 			method: request.method,
 			error: {
-				message: maybeParsedRequest.fullError === undefined ? 'Failed to parse RPC request ' : maybeParsedRequest.fullError.toString(),
+				message: `Failed to parse RPC request: ${ JSON.stringify(request) }`,
+				data: maybeParsedRequest.fullError === undefined ? 'Failed to parse RPC request' : maybeParsedRequest.fullError.toString(),
 				code: METAMASK_ERROR_FAILED_TO_PARSE_REQUEST,
 			}
 		}
 	}
```

Comment:
This can be simplified with some conditional inversion and combining two of the return paths.
```suggestion
if (maybeParsedRequest.success === false) {
	console.log(request)
	console.warn(maybeParsedRequest.fullError)
	const maybePartiallyParsedRequest = SupportedEthereumJsonRpcRequestMethods.safeParse(request)
	// the method is some method that we are not supporting, forward it to the wallet if signer is available
	if (maybePartiallyParsedRequest.success === false && forwardToSigner) return { forward: true as const, unknownMethod: true, ...request }
	return {
		method: request.method,
		error: {
			message: `Failed to parse RPC request: ${ JSON.stringify(request) }`,
			data: maybeParsedRequest.fullError === undefined ? 'Failed to parse RPC request' : maybeParsedRequest.fullError.toString(),
			code: METAMASK_ERROR_FAILED_TO_PARSE_REQUEST,
		}
	}
}
```